kernel/base/src/main/java/com/twosigma/beakerx/kernel/commands/MagicCommand.java

```diff
@@ -178,45 +183,80 @@ private MagicCommandFunctionality classpathAddJar() {
           throw new IllegalStateException("Wrong command format: " + CLASSPATH_ADD_JAR);
         }
 
-        this.kernel.addJarToClasspath(new PathToJar(split[3]));
+        String path = split[3];
+        if (doesPathContainsWildCards(path)) {
+          getPaths(path)
+              .forEach(currentPath -> kernel.addJarToClasspath(new PathToJar(currentPath)));
+        } else {
+          validatePath(path);
+          this.kernel.addJarToClasspath(new PathToJar(path));
+        }
+
         return getMagicCommandItem(code, message, executionCount);
       } catch (IllegalStateException e) {
         return new MagicCommandItemWithResult(
-                messageCreator
-                        .buildOutputMessage(message, e.getMessage(), true),
-                messageCreator.buildReplyWithoutStatus(message, executionCount)
+            messageCreator
+                .buildOutputMessage(message, e.getMessage(), true),
+            messageCreator.buildReplyWithoutStatus(message, executionCount)
         );
       }
     };
   }
 
+  private List<String> getPaths(String pathWithWildcard) {
+    String pattern = "**" + pathWithWildcard;
+    PathMatcher pathMatcher = FileSystems.getDefault().getPathMatcher("glob:" + pattern);
+    Path startPath = Paths.get("");
+
+    try {
+      List<String> paths = Files.find(startPath, Integer.MAX_VALUE,
+          (path, basicFileAttributes) -> pathMatcher.matches(path))
+          .filter(path -> path.toString().contains(".jar"))
```

Comment:
Could you provide more details? 

I allow to import only files that contains .jar